### ISSUE TITLE
[Question] Update multiple items in state in an array

### ISSUE THREAD

**Issue by Esya** [ngxs/store#1016]
Hi, 

We have an array in our state that we'll call "items" - these elements are displayed in a smart table.
We're displaying something like 700 rows at the moment. These items have a checkbox, called "checked" which is a boolean. We want the user to be able to select all, or unselect all items. 

We have no idea what the best way to update this property for all items inside the state is; we tried this : 

- Getting the array with getState
- Looping through all items of the array and calling an action that uses the `updateItem` stateOpeator

But this takes a few seconds. 

Looping through the items and changing the property directly returns an error as the property is readonly, which makes sense because of the immutability. We also tried to make up something using the scarcely documented `iif` state operator but to no success.

TLDR: What's the best way to quickly update hundreds of items in an array in the state?  

Thanks!

**Comment by mailok** [comment]
@Esya you cant try this: https://github.com/ngxs/store/issues/926#issuecomment-474417938

**Comment by Thenis** [comment]
700 items are not that much and should take less than a second to loop through them. Are you sure its not something else that is causing you performance problems?

**Comment by Esya** [comment]
Thanks, we used the `updateManyItems` custom state operator and that solved it for us. We think that this should be built-in, as this kind of operation seems trivial but the implementation is not.

**Comment by mailok** [comment]
@Esya I'm very glad that it was useful

**Comment by splincode** [comment]
@mailok Could you send us a pull request and add operator `updateManyItems`?

**Comment by mailok** [comment]
@splincode Yes of course

**Comment by mailok** [comment]
Hello @splincode , I am presenting problems with my internet and npm, I can not compile @ngxs, because my internet is very slow. I will not be able to do the Pull Request

**Comment by arturovt** [comment]
Closing this as this doesn't seem to be an "issue" :) You can ask this question in our [Slack](https://now-examples-slackin-eqzjxuxoem.now.sh/).
